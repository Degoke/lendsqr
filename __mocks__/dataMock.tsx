import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
