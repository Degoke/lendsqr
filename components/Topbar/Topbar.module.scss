@use '../../styles/variables' as *;

.topbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2vw;
  background-color: $white-background;
  box-shadow: 3px 0px 20px rgba(0, 0, 0, 0.04);
  position: sticky;
  top: 0;
  flex-wrap: wrap;
  gap: 1rem;
  z-index: 2;

  &__logoSection {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    gap: 8rem;

    input {
      border: 1px solid #213f7d;
      border-radius: 8px;
      height: 40px;
      width: 25rem;
      border-right: none;
      padding: 1rem;
    }

    @media all and (max-width: 800px) {
      input,
      button {
        display: none;
      }
    }

    button {
      height: 44px;
      background: #39cdcc;
      width: 56px;
      border-top-right-radius: 8px;
      border-bottom-right-radius: 8px;
      margin-left: -2rem;
      border: none;
    }
  }

  &__actionButtons {
    display: flex;
    align-items: center;
    gap: 1rem;
    justify-self: end;

    p {
      text-decoration: underline;
    }

    &_dropdown {
      display: flex;
      align-items: center;
      flex-wrap: wrap;
      font-size: 16px;
      font-weight: 500;
    }
  }

  @media all and (max-width: 800px) {
    &__actionButtons {
      display: none;
    }
  }
}

.dummyImage {
  border-radius: 50%;
}
