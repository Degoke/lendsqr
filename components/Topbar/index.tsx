import Image from 'next/image'
import Link from 'next/link'
import Logo from '../../public/images/logo.svg'
import SearchIcon from '../../public/images/search_icon.svg'
import DropdownIcon from '../../public/images/dropdown_icon.svg'
import DummyImage from '../../public/images/dummy_image.svg'
import NotificationIcon from '../../public/images/notification_icon.svg'
import IconButton from '../IconButton'
import styles from './Topbar.module.scss'

export default function Topbar() {
  return (
    <nav className={styles.topbar}>
      <div className={styles.topbar__logoSection}>
        <div>
          <Image src={Logo} alt="lendsqr logo" />
        </div>
        <form>
          <input type="search" placeholder="Search for something" />
          <button type="submit">
            <Image src={SearchIcon} alt="search button" />
          </button>
        </form>
      </div>
      <div className={styles.topbar__actionButtons}>
        <p>Docs</p>
        <IconButton image={NotificationIcon} alt="Notifications Button" />
        <Image
          className={styles.dummyImage}
          src={DummyImage}
          alt="user avatar"
        />
        <div className={styles.topbar__actionButtons_dropdown}>
          <span>Adedeji</span>
          <IconButton image={DropdownIcon} alt="user dropdown button" />
        </div>
      </div>
    </nav>
  )
}
