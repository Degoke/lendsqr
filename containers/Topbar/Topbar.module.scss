@use "../../styles/variables" as *;

.topbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1rem 2vw;
    background-color: $white-background;
    box-shadow: 3px 0px 20px rgba(0, 0, 0, 0.04);

    &__logoSection {
        display: flex;
        align-items: center;
    }

    &__actionButtons {
        display: flex;
        align-items: center;
    }
}