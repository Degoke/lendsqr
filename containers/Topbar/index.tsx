import Image from 'next/image'
import Link from 'next/link'
import Logo from '../../public/images/logo.svg'
import SearchIcon from '../../public/images/search_icon.svg'
import DropdownIcon from '../../public/images/dropdown_icon.svg'
import DummyImage from '../../public/images/dummy_image.svg'
import NotificationIcon from '../../public/images/notification_icon.svg'
import IconButton from '../../components/IconButton'
import styles from './Topbar.module.scss'


export default function Topbar() {
    return (
        <nav className={styles.topbar}>
            <div className={styles.topbar__logoSection}>
            <div>
                <Image src={Logo} alt="lendsqr logo" />
            </div>
            <form>
                <input type="search" />
                <button type="submit">
                    <Image src={SearchIcon} alt="search button" />
                </button>
            </form>
            </div>
            <div className={styles.topbar__actionButtons}>
                <Link href="/">
                    Docs
                </Link>
                <IconButton image={NotificationIcon} alt="Notifications Button" />
                <div>
                    <Image src={DummyImage} alt="user avatar" />
                </div>
                <div>
                    <span>Adedeji</span>
                    <IconButton image={DropdownIcon} alt="user dropdown button" />
                </div>
            </div>
        </nav>
    )
}