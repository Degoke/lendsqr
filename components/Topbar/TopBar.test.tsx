import { render, screen } from '@testing-library/react'
import Topbar from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('Topbar tests', () => {
  it('renders Topbar component unchanged', () => {
    const { container } = render(<Topbar />)
    expect(container).toMatchSnapshot()
  })
})
