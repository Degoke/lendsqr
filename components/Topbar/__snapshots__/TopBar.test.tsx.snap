// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar tests renders Topbar component unchanged 1`] = `
<div>
  <nav
    class="topbar"
  >
    <div
      class="topbar__logoSection"
    >
      <div>
        <img
          alt="lendsqr logo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <form>
        <input
          placeholder="Search for something"
          type="search"
        />
        <button
          type="submit"
        >
          <img
            alt="search button"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
      </form>
    </div>
    <div
      class="topbar__actionButtons"
    >
      <p>
        Docs
      </p>
      <button
        class="iconButton"
      >
        <img
          alt="Notifications Button"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </button>
      <img
        alt="user avatar"
        class="dummyImage"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <div
        class="topbar__actionButtons_dropdown"
      >
        <span>
          Adedeji
        </span>
        <button
          class="iconButton"
        >
          <img
            alt="user dropdown button"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
      </div>
    </div>
  </nav>
</div>
`;
